VSS.md

```diff
@@ -0,0 +1,518 @@
+
+The `FT.INFO` command returns information only for the current node.
+
+## Current Restrictions
+
+- `FT.SEARCH` can not be executed within a LUA function.
+- `FT.SEARCH` assumes that READONLY has been specified, meaning that broadcasts across the cluster may select either primaries or replicas for each shard.
+- The RDB file format is not compatible with Redisearch
+- ACLs are not supported
+
+
+### Commands
+
+```
+FT.CREATE <index-name>
+    ON HASH
+    [PREFIX <count> <prefix> [<prefix>...]]
+    SCHEMA 
+        (
+            <field-identifier> [AS <field-alias>] 
+                  NUMERIC 
+                | TAG [SEPARATOR <sep>] [CASESENSITIVE] 
+                | VECTOR [HNSW | FLAT] <attr_count> [<attribute_name> <attribute_value>]+)
+        )+
+```
+
+The `FT.CREATE` command creates an empty index and initiates the backfill process. Each index consists of a number of field definitions. Each field definition specifies a field name, a field type and a path within each indexed key to locate a value of the declared type. Some field type definitions have additional sub-type specifiers.
+
+For indexes on HASH keys, the path is the same as the hash member name. The optional `AS` clause can be used to rename the field if desired. 
+Renaming of fields is especially useful when the member name contains special characters.
+
+For indexes on JSON keys, the path is a JSON path to the data of the declared type. Because the JSON path always contains special characters, the `AS` clause is required.
+
+- **\<index-name\>** (required): This is the name you give to your index. If an index with the same name exists already, an error is returned.  
+    
+- **ON HASH | JSON** (optional): Only keys that match the specified type are included into this index. If omitted, HASH is assumed.  
+    
+- **PREFIX \<prefix-count\> \<prefix\>** (optional): If this clause is specified, then only keys that begin with the same bytes as one or more of the specified prefixes will be included into this index. If this clause is omitted, all keys of the correct type will be included. A zero-length prefix would also match all keys of the correct type.
+    
+- Field types:  
+    
+  - **TAG**: A tag field is a string that contains one or more tag values.  
+      
+    - **SEPARATOR \<sep\>** (optional): One of these characters `,.<>{}[]"':;!@#$%^&*()-+=~` used to delimit individual tags. If omitted the default value is `,`.  
+    - **CASESENSITIVE** (optional): If present, tag comparisons will be case sensitive. The default is that tag comparisons are NOT case sensitive
+
+  
+
+  - **NUMERIC** Field contains a number.  
+      
+  - **VECTOR**: Field contains a vector. Two vector indexing algorithms are currently supported: HNSW (Hierarchical Navigable Small World) and FLAT (brute force). Each algorithm has a set of additional attributes, some required and other optional.  
```

Comment:
Do we have a chance to support GRAPH_PQ ( combination of the HNSW algorithm and the PQ algorithm), IVF_GRAPH ( combination of IVF and HNSW) and IVF_GRAPH_PQ (combination of the PQ algorithm with the IVF or HNSW). These algorithms could work for huge amount records (from hundreds of millions to  more than 1 billion files in shards)

Reply:
We've taken the stance of putting into the RFC the functionality that's being contributed -- which doesn't include these particular algorithms. But with that said, there's no reason that this query architecture can't easily be extended to support these algorithms in the future.